ambry-cloud/src/main/java/com/github/ambry/cloud/BackupIntegrityMonitor.java

```diff
@@ -172,34 +235,88 @@ public void run() {
       /** Restore cloud backup C */
       logger.info("[BackupIntegrityMonitor] Restoring backup partition-{} to disk [{}]", partition.getId(), store);
       cloudReplica = azureReplicationManager.getCloudReplica(store.getReplicaId());
-      // TODO: Reload tokens
+      // No need to reload tokens, since we clear off disks before each run
       azureReplicator.addRemoteReplicaInfo(cloudReplica);
       RecoveryToken token = (RecoveryToken) cloudReplica.getToken();
       while (!token.isEndOfPartition()) {
         azureReplicator.replicate();
+        token = (RecoveryToken) cloudReplica.getToken();
         long numBlobs = token.getNumBlobs();
-        if (numBlobs > 0 && (numBlobs % RECOVERY_MILESTONE == 0)) {
-          // Print progress every N blobs
+        if (numBlobs > 0 && (numBlobs % azureConfig.azureBlobStorageMaxResultsPerPage == 0)) {
+          // Print progress, if N blobs have been restored from Azure
           logger.info("[BackupIntegrityMonitor] Recovered {} blobs {} bytes of partition-{} from Azure Storage",
               numBlobs, token.getBytesRead(), partition.getId());
         }
       }
-      logger.info("[BackupIntegrityMonitor] Restored backup partition-{} to disk [{}]", partition.getId(), store);
+      logger.info("[BackupIntegrityMonitor] Restored backup partition-{} to disk [{}], num_blobs = {}, num_bytes = {}",
+          partition.getId(), store, token.getNumBlobs(), token.getBytesRead());
 
-      /** TODO: Replicate from server and compare */
+      /** Create a temporary map of all keys recovered from cloud */
+      // Get all entries in one call, so keep the ceil buffer size high 1GB, all of it will not be used
+      FindInfo finfo = store.findEntriesSince(new StoreFindToken(), 1000 * (2 << 20),
```

Comment:
this will not give you all the entries in one call. The size you passed to the `findEntriesSince` is the chunk data size. A partition can have up to 384 GB of worth of data. I suggest you do a while loop to get all the entries.

Reply:
I see, how would I know if I've acquired a list of all keys in local store ?

Reply:
or what if pass an insanely large size like 1000 TiB ? will it return all keys in one call ?

Reply:
done